src/core/grid_scan_matcher.h

```diff
@@ -8,45 +14,97 @@
 #include "state_data.h"
 #include "sensor_data.h"
 
+/**
+ * \brief Interface of scan matcher observer.
+ */
 class GridScanMatcherObserver {
 public:
+  /**
+   * A callback that is invoked on scan matching start.
+   * \param RobotState Pose of a robot.
+   * \param TransformedLaserScan A laser scan with a transformation.
+   * \param GridMap A grid map that is used by the matcher.
+   */
   virtual void on_matching_start(const RobotState &,           /*pose*/
                                  const TransformedLaserScan &, /*scan*/
                                  const GridMap &) {}    /*map*/
+
+  /**
+   * A callback that is invoked on a corrected robot pose evaluation.
+   * \param RobotState Pose of a robot.
+   * \param TransformedLaserScan A laser scan with a transformation.
+   */
   virtual void on_scan_test(const RobotState &,           /*pose*/
                             const TransformedLaserScan &, /*scan*/
                             double) {};                  /*score*/
+
+  /**
+   * A callback that is invoked on updating a pose of robot.
```

Comment:
"on ..." -> "... when a corrected pose is better(?) than the  previously estimated one."